assets/src/onboarding-wizard/index.js

```diff
@@ -90,13 +95,24 @@ Providers.propTypes = {
 domReady( () => {
 	const root = document.getElementById( APP_ROOT_ID );
 
-	if ( root ) {
-		render(
-
-			<Providers>
-				<SetupWizard closeLink={ CLOSE_LINK } finishLink={ FINISH_LINK } appRoot={ root } />
-			</Providers>,
-			root,
-		);
+	if ( ! root ) {
+		return;
 	}
+
+	const errorHandler = ( error ) => {
+		// Handle only own errors.
+		if ( error?.filename?.match( /amp-onboarding-wizard(\.min)?\.js/ ) ) {
+			render( <ErrorScreen error={ error } />, root );
+		}
+	};
+
+	global.addEventListener( 'error', errorHandler );
+
+	render(
+		<Providers>
+			<SetupWizard closeLink={ CLOSE_LINK } finishLink={ FINISH_LINK } appRoot={ root } />
+		</Providers>,
+		root,
+		() => global.removeEventListener( 'error', errorHandler ),
```

Comment:
According to the [React docs](https://reactjs.org/docs/react-dom.html#render) the callback is executed after the component is rendered or updated, which means the error boundary will not be handling errors until all renders are complete. Maybe we could remove the listener in the `Providers` component?